### ISSUE TITLE
Bug: Release Artifact Download: Artifact not found, Falling back to parent directory results in unwanted behaviour

### ISSUE THREAD

**Issue by TMattes** [microsoft/azure-pipelines-tasks#14587]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DownloadPipelineArtifact  

## Environment
- Server - Azure Pipelines or TFS on-premises? **On-Premises**
    
    - If using TFS on-premises, provide the version: **ADO Server 2019.1.1**
    
- Agent - Hosted or Private: Private
    - If using private agent, provide the OS of the machine running the agent and the agent version: **Windows Server 2019, Agent version 2.177.0**

## Issue Description

In a release pipeline in the DownloadPipelineArtifact task, we specified a build from which to get the artifact. The artifact now is no longer on the file share, so the task falls back to parent directory and fetches everything it find there. Furthermore, it takes another artifact than the one we specified. Instead of that behaviour, it should fail to indicate, that the artifact could not be found. Otherwise, we are not able to rely on functionality of the pipeline and never know exactly, what is actually released.

**Please, please let the pipeline fail, if nothing is found in the specified directory! Please eliminate this useless fall-back!**

### Task logs

2021-03-12T07:18:22.3969559Z Downloading artifact <artifact name with verson string in it> from: <'File share an exact path to the wanted artifact incl. version number'> 
Directory '<'Artifact path'>' does not exist. Falling back to parent directory: <'parent path'>

## Troubleshooting

- The release pipeline is going on as if nothing went wrong and as it finds another artifact of the project in parent directory, it simply releases this. If we hadn't exactly checked what actually was processed, we would have released a wrong version of the product.

- Furthermore, it downloads everything in the parent directory which is a big amount of data not needed for this pipeline!

**Comment by TMattes** [comment]
Is there any update on this issue?

**Comment by TMattes** [comment]
I'm not very happy about your ignorance of this issue...

**Comment by TMattes** [comment]
When will this be finally fixed? On a regular basis, our building servers run out of space due to this issue!

**Comment by TMattes** [comment]
Still no action on this bug. :(

**Comment by TMattes** [comment]
This is not stale, but still very annoying! Please fix this bug!

**Comment by TMattes** [comment]
and again - please fix

**Comment by TMattes** [comment]
stale again, and nothing happened in more than 2 years. :(

**Comment by TMattes** [comment]
@AArnott @krukow @chandlerkent @GuyLewin 
Are you the guys who are able to fix this? If yes, why isn't there any action on this annoying bug? This crashes our build servers on a regular basis. I don't think, a big effort is necessary to fix this and let the download fail, if the artifact can't be found. Another Bug item (https://github.com/microsoft/azure-pipelines-tasks/issues/12306) about the same behaviour was automatically closed, although no action had happened on it at all.
Please fix this!

**Comment by AArnott** [comment]
Nope. Nothing I can do. I'm not on the Azure Pipelines team. Sorry.
